mini-e2e-operator.groovy

```diff
@@ -0,0 +1,234 @@
+				podman_login(ci_registry, '$CREDS_USER', '$CREDS_PASSWD')
+			}
+
+			// base_image is like quay.io/ceph/ceph:v19, strip "quay.io/"
+			podman_pull(ci_registry, "quay.io", "${base_image}" - q_io_regex)
+			// cephcsi:devel is used with 'make containerized-build'
+			podman_pull(ci_registry, ci_registry, "ceph-csi:devel")
+		}
+		stage('build artifacts') {
+			// build container image
+			ssh 'cd /opt/build/go/src/github.com/ceph/ceph-csi && make image-cephcsi GOARCH=amd64 CONTAINER_CMD=podman'
+			// build e2e.test executable
+			ssh "cd /opt/build/go/src/github.com/ceph/ceph-csi && make containerized-build CONTAINER_CMD=podman TARGET=e2e.test ENV_CSI_IMAGE_NAME=${ci_registry}/ceph-csi USE_PULLED_IMAGE=yes"
+		}
+		stage("deploy k8s-${k8s_version} and rook") {
+			def rook_version = sh(
+				script: 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} \'source /opt/build/go/src/github.com/ceph/ceph-csi/build.env && echo ${ROOK_VERSION}\'',
+				returnStdout: true
+			).trim()
+
+			if (rook_version != '') {
+				// single-node-k8s.sh pushes the image into minikube
+				podman_pull(ci_registry, "docker.io", "rook/ceph:${rook_version}")
+			}
+
+			def rook_ceph_cluster_image = sh(
+				script: 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} \'source /opt/build/go/src/github.com/ceph/ceph-csi/build.env && echo ${ROOK_CEPH_CLUSTER_IMAGE}\'',
+				returnStdout: true
+			).trim()
+			def q_io_regex = ~"^quay.io/"
+
+			if (rook_ceph_cluster_image != '') {
+				// single-node-k8s.sh pushes the image into minikube
+				podman_pull(ci_registry, "quay.io", rook_ceph_cluster_image - q_io_regex)
+			}
+
+			timeout(time: 30, unit: 'MINUTES') {
+				ssh "./single-node-k8s.sh --k8s-version=${k8s_release}"
+			}
+
+			// vault:latest,1.8.5 and nginx:latest are used by the e2e tests
+			podman_pull(ci_registry, "docker.io", "library/vault:latest")
+			ssh "./podman2minikube.sh docker.io/library/vault:latest"
+			podman_pull(ci_registry, "docker.io", "library/nginx:latest")
+			ssh "./podman2minikube.sh docker.io/library/nginx:latest"
+			podman_pull(ci_registry, "docker.io", "library/vault:1.8.5")
+			ssh "./podman2minikube.sh docker.io/library/vault:1.8.5"
+		}
+		stage('deploy ceph-csi through operator') {
+			def operator_version = sh(
+				script: 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} \'source /opt/build/go/src/github.com/ceph/ceph-csi/build.env && echo ${CEPH_CSI_OPERATOR_VERSION}\'',
+				returnStdout: true
+			).trim()
+
+			podman_pull(ci_registry, "quay.io", "cephcsi/ceph-csi-operator:${operator_version}")
+			ssh "./podman2minikube.sh quay.io/cephcsi/ceph-csi-operator:${operator_version}"
+
+			timeout(time: 30, unit: 'MINUTES') {
+				ssh 'cd /opt/build/go/src/github.com/ceph/ceph-csi && ./scripts/deploy-ceph-csi-operator.sh deploy'
```

Comment:
do you have an other PR that contains this script?

Reply:
Yes, I'll be raising the PR for the script.